routes/api/project/location/private_subnet.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class CloverApi
+  hash_branch(:project_location_prefix, "private-subnet") do |r|
+    @serializer = Serializers::Api::PrivateSubnet
+
+    r.get true do
+      result = @project.private_subnets_dataset.where(location: @location).authorized(@current_user.id, "PrivateSubnet:view").eager(:nics).paginated_result(
```

Comment:
Since nic serializer uses its private subnet
```suggestion
      result = @project.private_subnets_dataset.where(location: @location).authorized(@current_user.id, "PrivateSubnet:view").eager(nics: [:private_subnet]).paginated_result(
```